src/key_value_store/core/KeyValueStore.cpp

```diff
@@ -0,0 +1,66 @@
+#include "KeyValueStore.h"
+
+#include <mutex>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+// Alias for the unorderedList
+typedef std::unordered_map<std::string, 
+  std::vector<std::string>> hashMap; 
+
+bool KeyValueStore::Put(std::string key, std::string value) {
+  bool result; 
+  lock_.lock();
```

Comment:
Better to use unique_lock or lock_guard

Reply:
Shifted to lock_guard instead of unique_lock because I do not need to lock, unlock and re-lock in the same function. 

I read that lock_guard "Allow simultaneous lock of multiple mutexes to overcome deadlock problem." Wouldn't this be counterproductive to locking? If multiple threads can lock the resource and read/write to it? Can they act as semaphores?

Reply:
lock_guard should be fine in this setting -- the deadlock avoidance mechanism simply makes sure if multiple locks are held that the are acquired / released in the same order across threads.